### ISSUE TITLE
Salsa duplicate literal properties cause should be flagged only when in strict mode

### ISSUE THREAD

**Issue by egamma** [Microsoft/TypeScript#6650]
_From @alexandrudima on January 27, 2016 13:53_

Testing #2218 

The duplicate here is not a problem because the code is not in strict mode (also tried running it and node runs it just fine):

```javascript
// "use strict";
var o = { p: 1, p: 2};
console.log(o);
```
![image](https://cloud.githubusercontent.com/assets/5047891/12615335/6e88aee4-c505-11e5-8a8e-81ae41632fff.png)

## But in ES6 this is not an error anymore?

I have also found the following interesting note at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode:
![image](https://cloud.githubusercontent.com/assets/5047891/12615377/99c5e77a-c505-11e5-90a7-1660fbf5216d.png)

_Copied from original issue: Microsoft/vscode#2452_

**Comment by egamma** [comment]
_From @alexandrudima on January 27, 2016 13:55_

The same is true regarding other flagged errors that should only be errors in strict mode:

![image](https://cloud.githubusercontent.com/assets/5047891/12615440/0151d5de-c506-11e5-901b-abf23f3a7263.png)

**Comment by billti** [comment]
It's not an error per the spec, but we flag this as almost a linter type check, as it is nearly always unintended. 

Some of the other bugs logged are where you have asked for linter like checks where there is no error per the spec. Is this something you want us to suppress? Else this is 'by design'.

**Comment by alexandrudima** [comment]
Works for me :+1:

**Comment by mhegazy** [comment]
The error is already generated, but filtered out. as part of #6802, this should be enabled again.
